backend/dataall/modules/s3_datasets_shares/services/share_managers/lf_share_manager.py

```diff
@@ -1,31 +1,71 @@
-from dataall.base.aws.quicksight import QuicksightClient
+from enum import Enum, auto
+from typing import List
+
 from dataall.base.aws.iam import IAM
+from dataall.base.aws.quicksight import QuicksightClient
 from dataall.base.aws.sts import SessionHelper
 from dataall.base.db import exceptions
+from dataall.core.environment.services.environment_service import EnvironmentService
+from dataall.modules.s3_datasets.db.dataset_models import DatasetTable
+from dataall.modules.s3_datasets_shares.aws.glue_client import GlueClient
+from dataall.modules.s3_datasets_shares.aws.lakeformation_client import LakeFormationClient
+from dataall.modules.s3_datasets_shares.services.s3_share_alarm_service import S3ShareAlarmService
+from dataall.modules.shares_base.db.share_object_models import ShareObjectItem, ShareObjectItemDataFilter
 from dataall.modules.shares_base.db.share_object_repositories import ShareObjectRepository
 from dataall.modules.shares_base.db.share_object_state_machines import ShareItemSM
 from dataall.modules.shares_base.db.share_state_machines_repositories import ShareStatusRepository
+from dataall.modules.shares_base.services.share_manager_utils import ShareErrorFormatter
 from dataall.modules.shares_base.services.shares_enums import (
     ShareItemStatus,
     ShareObjectActions,
     ShareItemActions,
     ShareItemHealthStatus,
+    ShareObjectDataPermission,
 )
-from dataall.modules.s3_datasets.db.dataset_models import DatasetTable
-from dataall.modules.s3_datasets_shares.services.s3_share_alarm_service import S3ShareAlarmService
-from dataall.modules.shares_base.db.share_object_models import ShareObjectItem, ShareObjectItemDataFilter
-from dataall.modules.shares_base.services.share_manager_utils import ShareErrorFormatter
 from dataall.modules.shares_base.services.sharing_service import ShareData
 
 logger = logging.getLogger(__name__)
 
 
+class LfPermType(Enum):
+    Table = auto()
+    Database = auto()
+    ResourceLink = auto()
+    Filters = auto()
+
+
+PERM_TO_LF_PERMS = {
+    ShareObjectDataPermission.Read.value: {
+        LfPermType.Table: ['DESCRIBE', 'SELECT'],
+        LfPermType.Database: ['DESCRIBE'],
+        LfPermType.ResourceLink: ['DESCRIBE'],
+        LfPermType.Filters: ['SELECT'],
+    },
+    ShareObjectDataPermission.Write.value: {
+        LfPermType.Table: ['INSERT'],
+        LfPermType.Database: ['CREATE_TABLE'],
+        LfPermType.ResourceLink: ['DESCRIBE'],
+        LfPermType.Filters: ['SELECT'],
+    },
+    ShareObjectDataPermission.Modify.value: {
+        LfPermType.Table: ['ALTER', 'DROP', 'DELETE'],
+        LfPermType.Database: ['ALTER', 'DROP'],
+        LfPermType.ResourceLink: ['DESCRIBE'],
+        LfPermType.Filters: ['SELECT'],
+    },
+}
+
+
+def perms_to_lfperms(permissions: List[str], lf_perm_type: LfPermType) -> List[str]:
+    lfperms = list()
+    for p in permissions:
+        lfperms.extend(PERM_TO_LF_PERMS[p][lf_perm_type])
+    return list(dict.fromkeys(lfperms))
```

Comment:
It is for dedup yes and this approach is used because a `set` doesn't preserve the order while `dict.fromkeys` does. Although order doesn't make any logical difference it does help with testing (i.e think `assert_called_once_with(['P1', 'P2])` etc)